### ISSUE TITLE
feature request: Allow `EpicResult` to assign `Promise<ActionLike>` and implement built-in Action: `ignore`

### ISSUE THREAD

**Issue by sisisin** [typeless-js/typeless#16]
### Feature request

- Allow `EpicResult` to assign `Promise<ActionLike>`
- Implement built-in Action: `TypelessActions.ignore()`

#### Allow `EpicResult` to assign `Promise<ActionLike>`

async/await is sometimes useful.

Following example, make an Action with `getSomeId`'s result and `getSomebodyById`'s result using `Observable` pattern and `Promise` pattern

```typescript
  .epic()
  // using Observable
  .on(SomeActions.fetchSomebody, () => {
    return Rx.from(API.getSomeId()).pipe(
      Rx.mergeMap(id =>
        Rx.from(API.getSomebodyById(id)).pipe(
          Rx.map(somebody => SomeActions.fetchSomebodyFulfilled({ id, somebody })),
        ),
      ),
      Rx.catchError(e => Rx.of(SomeActions.rejected())),
    );
  })
  // using Promise
  .on(SomeActions.fetchSomebody, async () => {
    try {
      const id = await API.getSomeId();
      const somebody = await API.getSomebodyById(id);
      return SomeActions.fetchSomebodyFulfilled({ id, somebody });
    } catch (e) {
      return SomeActions.rejected();
    }
  });
```

Nested `Observable` is too complex.
So, I want to use `Promise` like in this case.

#### Implement built-in Action: `TypelessActions.ignore()`

It is difficult to ignore `EpicHandler` which returns `Promise`.
So, create built-in `ignore` Action.

```typescript
  .on(SomeActions.fetchSomebody, async () => {
    // this result type is `Promise<Observable<>>`, so this is not work!
    return Rx.empty();
  })
  .on(SomeActions.fetchSomebody, async () => {
    // this result type is Promise<ActionLike>, and works well!
    return TypelessActions.ignore();
  })
```

When returned `TypelessActions.ignore()` , it is converted `Rx.empty()` in `createOutputStream`.

---

I think this feature request may breaks typeless concept.
So, If you say this is not for typeless, I'll certainly withdraw this request.

Thank you.

**Comment by BetterCallSky** [comment]
Hello @sisisin 
Thanks for your suggestion. I am fine with supporting promises because it doesn't add much complexity, but I am not sure about `TypelessActions.ignore()`.
Currently, `TypelessActions.ignore()`, will be logged in the console and other epics could listen for it etc. I think it would be much easier to just return a `null` value?

**Comment by sisisin** [comment]
@BetterCallSky Thanks for reply.
I'm sure for your opinion, I think return a `null` for ignoring action is good!
I'll fix PullRequest.

**Comment by sisisin** [comment]
resolved at #17
